src/lib/matcher.rs

```diff
@@ -407,3 +408,252 @@ impl MatcherFactory {
+    #[case(vec![("AGG".as_bytes().to_vec()), ("C".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "GGGGG".as_bytes(), bitvec![0, 0, 0, 0, 0])] // fixed pattern set with multiple patterns and no matches
+    #[case(vec![("AGG".as_bytes().to_vec()), ("C".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGG".as_bytes(), bitvec![1, 1, 1])] // fixed pattern set where 1/3 patterns have matches
+    #[case(vec![("AG".as_bytes().to_vec()), ("C".as_bytes().to_vec()), ("TT".as_bytes().to_vec())], "AGACGTT".as_bytes(), bitvec![1, 1, 0, 1, 0, 1, 1])] // fixed pattern set with multiple matches
+    #[case(vec![("GAGA".as_bytes().to_vec()),("AGTT".as_bytes().to_vec())], "GAGAGTT".as_bytes(), bitvec![1, 1, 1, 1, 1, 1, 1])] // fixed pattern set with overlapping matches
+    fn test_to_bitvec_fixed_string_set(
+        #[case] patterns: Vec<Vec<u8>>,
+        #[case] bases: &[u8],
+        #[case] expected: BitVec,
+    ) {
+        let opts = MatcherOpts {
+            invert_match: false,
+            reverse_complement: false,
+            color: false,
+        };
+
+        let matcher = FixedStringSetMatcher {
+            patterns: patterns,
+            opts: opts,
+        };
+
+        let ranges = matcher.patterns.iter().flat_map(|pattern| {
+            bases
+                .find_iter(&pattern)
+                .map(|start| Range {
+                    start,
+                    end: start + pattern.len(),
+                })
+                .collect::<Vec<_>>()
+        });
+
+        let result_bitvec = to_bitvec(ranges.into_iter(), bases.len());
+        assert_eq!(result_bitvec, expected);
+    }
+
+    // ############################################################################################
+    // Tests to_bitvec() regex
+    // ############################################################################################
+    #[rstest]
+    #[case("^A", "AGG".as_bytes(), bitvec![1, 0, 0])] // regex with one match
+    #[case("^T", "AGG".as_bytes(), bitvec![0, 0, 0])] // regex with no matches
+    #[case("A.T", "ATTTGGGATT".as_bytes(), bitvec![1, 1, 1, 0, 0, 0, 0, 1, 1, 1])] // regex with multiple non overlapping matches
+    #[case("A.A", "ATATA".as_bytes(), bitvec![1, 1, 1, 0, 0])] // regex with overlapping matches
+    fn test_regex_bitvec(#[case] pattern: &str, #[case] bases: &[u8], #[case] expected: BitVec) {
+        let opts = MatcherOpts {
+            invert_match: false,
+            reverse_complement: false,
+            color: false,
+        };
+        let matcher = RegexMatcher::new(&pattern, opts);
+        let ranges = matcher.regex.find_iter(bases).map(|m| m.range());
+        let result_bitvec = to_bitvec(ranges, bases.len());
+        println!("{:?}", &result_bitvec);
+        assert_eq!(result_bitvec, expected);
+    }
+
+    // ############################################################################################
+    // Tests to_bitvec() regex set
```

Comment:
Why test both together?  or are you using `to_bitvec` to make sure it matches the correct range of bases?  If the latter, I think that's fine, but we should have our own `to_bitvec` tests that do not rely on matchers.